config/zfs-build.m4

```diff
@@ -49,6 +49,11 @@ AC_DEFUN([ZFS_AC_DEBUG], [
 		[ZFS_AC_DEBUG_DISABLE],
 		[AC_MSG_ERROR([Unknown option $enable_debug])])
 
+	AS_CASE(["x$FORCE_DEBUG"],
```

Comment:
We could just check if enable_invariants is set to yes and avoid the changes in ZFS_AC_DEBUG_INVARIANTS.

Reply:
This is true, I just was opting against using the same check in two different places.

I'll go do it and push.

Reply:
That's done. It seems to DTRT on my 14-CURRENT testbed with GENERIC and GENERIC-NODEBUG.